models/sources/sources.sql

```diff
@@ -0,0 +1,19 @@
+/* Bigquery won't let us `where` without `from` so we use this workaround */
+with dummy_cte as (
+    select 1 as foo
+)
+
+select
+    cast(null as {{ type_string() }}) as command_invocation_id,
+    cast(null as {{ type_string() }}) as node_id,
+    cast(null as {{ type_timestamp() }}) as run_started_at,
+    cast(null as {{ type_string() }}) as database,
+    cast(null as {{ type_string() }}) as schema,
+    cast(null as {{ type_string() }}) as source_name,
+    cast(null as {{ type_string() }}) as loader,
+    cast(null as {{ type_string() }}) as name,
+    cast(null as {{ type_string() }}) as identifier,
+    cast(null as {{ type_string() }}) as loaded_at_field,
+    cast(null as {{ type_json() }}) as freshness
```

Comment:
This was an ARRAY type before for Snowflake which is currently leading to errors when I do:
```dbt run -m dbt_artifacts```

when I install this branch into a project which already is using dbt_artifacts:

```
10:11:48  Database Error in model sources (models/sources/sources.sql)
10:11:48    002023 (22000): SQL compilation error:
10:11:48    Expression type does not match column data type, expecting ARRAY but got OBJECT for column FRESHNESS
10:11:48    compiled SQL at target/run/dbt_artifacts/models/sources/sources.sql`
```

Reply:
Hmm, this will be tricky to solve in a way that's compatible to both Snowflake and Bigquery I think, given that one uses JSON and the other ARRAY

https://github.com/brooklyn-data/dbt_artifacts/blob/9146d71d2ec459f549098bc52cab3cb2aa7eadb2/macros/create_sources_table_if_not_exists.sql#L36-L66

One option would be to create a specific type helper for just this column, but that seems a bit suboptimal IMO.  Do you have any ideas @NiallRees?

Reply:
Maybe we could offer a one-time migration for the source tables and cut a new major version with this new method?

Reply:
Think I'd be inclined to use a specific type helper (just doing an if snowflake else in the model vs a macro) rather than adding more complexity to the migration process.

Reply:
@NiallRees done and ready for another test.